runtime/src/main/java/org/corfudb/runtime/LogReplicationClient.java

```diff
@@ -0,0 +1,277 @@
+                            if (clientDestinations.getDestinationIdsList().contains(dest)) {
+                                return false;
+                            }
+                        }
+
+                        clientDestinations = clientDestinations.toBuilder().addAllDestinationIds(remoteDestinations).build();
+                    }
+                    else {
+                        clientDestinations = SinksInfoVal.newBuilder().addAllDestinationIds(remoteDestinations).build();
+                    }
+
+                    txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinations, null);
+                    txn.commit();
+
+                    return true;
+                }
+                catch (Exception e) {
+                    log.error("Unable to add destinations", e);
+                    return false;
+                }
+            }).setOptions(x -> x.setMaxRetryThreshold(Duration.ofSeconds(SYNC_THRESHOLD))).run();
+        } catch (InterruptedException e) {
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Remove destination from LogReplicationSourceMetadataTable
+     *
+     * @param logicalGroup String representative of the group of tables which are replicated
+     * @param destination Cluster from which the logicalGroup should be removed from further replication
+     * @return true if destination was removed
+     */
+    public boolean removeDestination(String logicalGroup, String destination) {
+        try {
+            return IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+
+                    ClientInfoKey clientInfoKey = ClientInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    if (txn.isExists(LR_SOURCE_METADATA_TABLE_NAME, clientInfoKey)) {
+                        SinksInfoVal clientDestinationIds = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+                        List<String> clientDestinationIdsList = new ArrayList<>(clientDestinationIds.getDestinationIdsList());
+
+                        if (clientDestinationIdsList.remove(destination)) {
+                            clientDestinationIds = clientDestinationIds.toBuilder()
+                                    .clearDestinationIds()
+                                    .addAllDestinationIds(clientDestinationIdsList)
+                                    .build();
+                            txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinationIds, null);
+                            txn.commit();
+                            return true;
+                        }
+                        return false;
```

Comment:
a debug logging would be nice here that says the destination was not found in the list